src/main/java/yeelp/scalingfeast/handlers/EnchantmentHandler.java

```diff
@@ -49,15 +50,16 @@ public void onAttackEvent(LivingAttackEvent evt)
 			int level = EnchantmentHelper.getMaxEnchantmentLevel(SFEnchantments.famine, (EntityLivingBase) evt.getSource().getTrueSource());
 			if(level != 0)
 			{
-				//20 ticks a second, so duration should be multiplied by 20
-				int hduration = level*20;
-				int wlevel = (level < 3 ? 0 : 1);
-				//level is also zero indexed.
-				if(entity instanceof EntityPlayer)
+				if(!(entity instanceof EntityPlayer))
 				{
-					entity.addPotionEffect(new PotionEffect(MobEffects.HUNGER, hduration, 50));
+					int wlevel = (level < 3 ? 0 : 1);
+					entity.addPotionEffect(new PotionEffect(MobEffects.WEAKNESS, 15*20,  wlevel));
 				}
-				entity.addPotionEffect(new PotionEffect(MobEffects.WEAKNESS, 15*20,  wlevel));
+				else
+				{
+					EntityPlayer player = (EntityPlayer) entity;
+					HungerDamage.damageFoodStats(player, 2.25f*level);
```

Comment:
The Hunger effect deducts 1 food point (a combination of hunger and saturation) from players every 40 seconds at level 1 (4.0 max exhaustion / 0.005 exhaustion per tick * 20 ticks per second). Higher levels scale proportionally.

At Hunger level 50, the Hunger effect deducts 1.25 food points from players per second. The Famine enchantment applies that effect level of the Hunger effect for 1 second per enchantment level, deducting 1.25 food points in the process.

The code you've implemented here deducts **2.25** food points from players per hit per enchantment level (since `HungerDamage.damageFoodStats` deducts the number given in food points). This is 1.00 food points **more** than the 1.25 food points per enchantment level I've calculated earlier. Why is it that way?

Reply:
This isn't necessarily final, but it was something I wanted to experiment with. The loss in hunger isn't instantaneous, as the game will deduct 1 hunger/saturation per tick while the player's exhaustion >= max exhaustion (which is by default 4.0), which is how it's done here.

In addition, I want to check to see if it's possible for Famine to only activate on full swings (so the player can't spam click). So the 2.25 was also done preemptively to compensate for this, as the windup time for swings and the way hunger will deplete hard caps the amount of hunger you can drain per second.

I hadn't gotten around to testing it yet so I'm not sure how all these dynamics play out yet.